src/compiler/scala/tools/nsc/typechecker/Unapplies.scala

```diff
@@ -128,7 +128,11 @@ trait Unapplies extends ast.TreeDSL {
    */
   def factoryMeth(mods: Modifiers, name: TermName, cdef: ClassDef): DefDef = {
     val tparams   = constrTparamsInvariant(cdef)
-    val cparamss  = constrParamss(cdef)
+    // don't propagate e.g. @volatile annot to apply's argument
+    def valDefWithoutAnnots(vd: ValDef): ValDef =
+      treeCopy.ValDef(vd, vd.mods.mapAnnotations(_ => Nil), vd.name, vd.tpt, vd.rhs)
```

Comment:
good catch! I have a more robust approach that drops non-param-targeting annots in the info completer